cpp/dolfinx/geometry/GJK.h

```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2020 Chris Richardson
+//
+// This file is part of DOLFINX (https://www.fenicsproject.org)
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#pragma once
+
+#include <Eigen/Dense>
+
+namespace dolfinx
+{
+namespace geometry
+{
+
+/// Calculate the distance between two convex bodies bd1 and bd2
+/// @param[in] p Body 1 list of vertices
+/// @param[in] q Body 2 list of vertices
+/// @return shortest vector between bodies
+Eigen::Vector3d
+gjk_vector(const Eigen::Matrix<double, Eigen::Dynamic, 3, Eigen::RowMajor>& p,
```

Comment:
Function name is cryptic.

Reply:
Is `compute_distance_between_bodies` better?

Reply:
Or maybe `compute_distance_gjk` - a bit shorter

Reply:
But is it a distance? Isn't that implying a scalar result?

Reply:
True. But what is it? Docstring says distance, return type is a vector . . . Is a vector useful if we don't know the closest point on one of the bodies?

Reply:
often one body is a point, so then it is useful, I think.